src/core/unit.c

```diff
@@ -1474,6 +1478,17 @@ int unit_load_fragment_and_dropin(Unit *u, bool fragment_required) {
                         u->source_mtime = 0;
         }
 
+        for (size_t i = 0; i < _UNIT_DEPENDENCY_MAX; i++) {
+                char **dep_names = u->dep_names_list[i];
+                if (!strv_isempty(dep_names)) {
+                        STRV_FOREACH(n, dep_names) {
+                                r = unit_add_dependency_by_name(u, i, *n, true, UNIT_DEPENDENCY_FILE);
```

Comment:
For boolean arguments, please add an inline comment with the name of the argument